---
author:
  - name: Stuart Lee
    affiliation: Monash University
    # use this syntax to add text on several lines
    address: |
      | Department of Econometrics and Business Statistics, Monash University
    email: stuart.a.lee@monash.edu
  - name: Dianne Cook
    address: |
      | Department of Econometrics and Business Statistics, Monash University
    affiliation: Monash University
    email: dicook@monash.edu
  - name: Natalia Da Silva
    address: |
      | Instituto de Estadística (IESTA), Universidad de la República
    affiliation: Universidad de la República
    email: natalia@iesta.edu.uy
  - name: Ursula Laa 
    affiliation: University of Natural Resources and Life Sciences
    address:  |
      | Institute of Statistics, University of Natural Resources and Life Sciences
    email:  ursula.laa@boku.ac.at
  - name: Nicholas Spyrison
    address: |
      | Faculty of Information and Technology, Monash University
    affiliation: Monash University
    email: nicholas.spyrison@monash.edu
  - name: Earo Wang
    address: |
      | Department of Statistics, The University of Auckland
    affiliation: The University of Auckland
    email: earo.wang@auckland.ac.nz
  - name: H. Sherry Zhang
    address: |
      | Department of Econometrics and Business Statistics, Monash University
    affiliation: Monash University
    email: huize.zhang@monash.edu
title:
  formatted: "A Review of the State-of-the-Art on Tours for Dynamic Visualisation of High-dimensional Data"
abstract: > 
  This article discusses a high-dimensional visualization technique called the tour, which can be used to view data in more than three dimensions. We review the theory and history behind the technique, as well as modern software developments and applications of the tour that are being found across the sciences and machine learning. 
keywords:
  formatted: ["tours", "data visualization", "high-dimensional data", "data science", "exploratory data analysis"]
output: 
  bookdown::pdf_document2:
    template: template/wires_template_rmd.tex
    keep_tex: true
    toc: false
bibliography: references.bib
biblio-style: apalike
---
```{r setup, include = FALSE}
knitr::opts_chunk$set(
  echo = FALSE,
  message = FALSE,
  warning = FALSE,
  fig.align = "center",
  fig.show = 'hold',
  dpi = 300 #,
  # out.width = "\\textwidth",
  # fig.path = here::here('img/')
)
```

<!--
Writing style: third person (it was done, ...)
-->

# Introduction

<!-- Purpose/why and what is high-d data (Di)-->

Data commonly arrives with more than two measured variables, which makes it more complicated to plot on a page. With multiple variables, especially if there is some association between variables, this would be called multivariate or high-dimensional data. When the variables are all numeric, or quantitative, visualisation often relies on some form of dimension reduction. This can be done by taking linear projections, for example, principal component analysis [@Hotelling1933-of] or linear discriminant analysis [@Fisher1936]. It is also common to reduce dimension nonlinearly with techniques like multidimensional scaling (MDS) [@Kruskal1964-do] or t-Distributed Stochastic Neighbour Embedding (t-SNE) [@Van_der_Maaten2008-qa]. 

The term "high-dimensional" here means Euclidean space. Figure \ref{fig:cubes} shows a way to imagine this. It shows a sequence of cube wireframes, ranging from 1D through to 6D, where beyond 2D is a linear projection of the cube. As dimension increase, a new orthogonal axis is added. For cubes, this is achieved by doubling the cube: a 2D is two 1D cubes, a 3D is two 2D cubes, and so forth. 

\begin{figure*}[!h]
\centerline{\includegraphics[width=0.95\textwidth]{figures/cubes.png}}
\caption{Illustrating what is meant by "high-dimensional" in this paper, and a "linear projection". From a sequence of increasing dimension cubes, from 1D to 6D, as wireframes, it can be seen that the as dimension increase by 1, the cube doubles. (Note to Di: to change this diagram to highlight the doubling.)}
\label{fig:cubes}
\end{figure*}

The focus of this review will be on visualising high-dimensional numerical data using linear projections, in particular, as provided by the grand tour [@Asimov1985-xr; @Buja2005-cx]. The reason being that it is not feasible to adequately review the very large area of visualising high-dimensions, and there have been numerous developments in tours recently. An overview of the technique, and new modifications is provided, along with how these techniques can be used in a variety of applications, and to help understand nonlinear dimension reduction results.

While tours are invaluable for assessing the geometry of data, they are by no means the only technique available for visualising structure in high dimensional data. An early technique proposed for assessing pairwise relationships between variables is the scatter plot matrix (SPLOM) [@Tukey1981-ov; @Chambers1983-dh; @Tukey1983-fj; @Carr1984-xw; @Becker1987-dw]. The SPLOM allows the viewer to assess correlation structure but does not scale to large numbers of variables. Similarly, parallel coordinates plots (PCP) can be used to explore correlation and multicollinearity [@Inselberg1985-kf; @Wegman1990-dy]. By placing multiple variables side by side in a PCP higher order structure like clustering or lower dimensional embeddings is revealed, however the ordering of variables along the axis changes what can be learned. Another display that relies on variable ordering is the heatmap, which is widely used to visualise cluster structure in bioinformatics. @Wilkinson2009-pj provides a comprehensive history of this display in the social and natural sciences.

A tour can be considered to be a dynamic graphic, because it shows a smooth sequence of projections over time, ideally with controls that allow stopping, reversing, changing direction, or going forward again. It can be useful to embed a tour into an interactive graphics system, where plots can be queried and elements highlighted (see for e.g. @Swayne2003-cc or @Tierney1991). To create the smooth sequence, a geodesic interpolation is computed between consecutive frames.  It allows the viewer to extrapolate from the low-dimensional to  shapes corresponding to multivariate distribution, and is particularly useful for detecting clusters, outliers and non-linear dependence. 

Nonlinear dimension reduction techniques such as t-SNE have become very popular in recent years, primarily for the ability to capture cluster structure in a succinct visual summary. However, it is only a summary, and it likely involves substantial warping of the original data space. Using the tour along with these techniques can illuminate the nature of the warped space, and reveal other structure lost in the dimension reduction.  Figure \ref{fig:tsne-tour} illustrates the difference in what can be learned from a tour in comparison with nonlinear dimension reduction using t-SNE. The 10-dimensional data comes from @Rauber2009-kk. The t-SNE view (plot A) shows six clearly separated clusters, all spherical with different sizes. The tour shows that the clusters look very little like this [EW: not sure what "like this" referes to] in the full data space. The clusters are at various distances apart and very different sizes. In the four projections from the tour this can be seen. The two green clusters are large and spherical, and far from the brown clusters. The brown clusters have one larger one, and three smaller, very close to each other. All of these are elliptical, which means that they have very little variability, actually no variability if you watch the full tour, in some of the 10 dimensions. This gives some deeper perspective to what is learned from t-SNE, and illustrates what the t-SNE dimension reduction has done: it has found small gaps between points and expanded these gaps to yield the representation. It should be noted, though, that the t-SNE view clearly reveals the presence of six distinct clusters. With this information is the invitation to look closer at the data in the tour, to see, that yes, indeed, there are three, tiny, tiny clusters very close to each other.

\begin{figure*}[!t]
         \includegraphics[width=0.5\textwidth]{figures/cluster_example_r.pdf}
     \vfill
   \begin{minipage}{\linewidth}     
   \centering
         \includegraphics[width=0.24\textwidth]{figures/cluster_example_tour1_r.pdf}
         \includegraphics[width=0.24\textwidth]{figures/cluster_example_tour2_r.pdf}
         \includegraphics[width=0.24\textwidth]{figures/cluster_example_tour3_r.pdf}
         \includegraphics[width=0.24\textwidth]{figures/cluster_example_tour4_r.pdf}
   \end{minipage}%
\caption{Comparison of structure perception between nonlinear dimension reduction using t-SNE (A) and projections seen during a tour (B.1-4). There are six clusters, as seen in the t-SNE view, but the relative distance between the clusters is extremely varied. This can be seen in the tour projections. The two green clusters are spherical in shape, and very distance from the brown clusters. Three of the brown clusters are very close to each other (just visible in B.1), and all brown clusters are elliptical. The tour provides a more accurate rendering of the clusters in the high-dimensional space, and complements what is learned from the dimension reduction.}
\label{fig:tsne-tour}
\end{figure*}


The rest of the review is structured as follows: in section \@ref(notation) we [EW: all of a sudden, from third person to first person?] define the notation and components of a tour display. By its nature a tour is most effective to analyst when combined with interactivity; in section \@ref(interaction) we review the components of user interfaces for manipulating tour views. The implementation of tour paths in statistical software is reviewed in section \@ref(software), where we discuss the history and state of the art packages for computing them. Section \@ref(applications) shows the diverse applications of the tour in the natural sciences, machine learning and applied statistics. Finally, in section \@ref(discussion) we point to future research directions for tours. 

# Tours for high dimensional visualization {#notation}

## Notation 

Below introduces the notations used for describing $d$-dimensional projections in a $p$-dimensional space. Let $X_{n \times p}$ be the data whose projections are of interest. A projection basis, $A_{p \times d}$, characterizes the direction from which the data are projected and needs to satisfy an orthonormality condition, which requires each column to be a unit vector and to be perpendicular to each other. This condition gives the $p$-dimensional basis space a geometrical structure of a Stiefel manifold. With a data matrix and a projection basis, a projection of the data can be defined as $Y = X \cdot A$. Figure \ref{fig:notation-huber} shows an example of the 2D iris data (sepal measures only) in a huber's plot, along with a 1D projection in the histogram. This projection is created based on projecting the data onto the solid line in the huber's plot and it separates group setosa from versicolor and virginica. (This plot is created with package PPtreeViz, whose reference is @Lee2005-gl)

\begin{figure*}
   \centering
         \includegraphics[width=\textwidth]{figures/notation-huber.pdf}
\caption{Huber's plot (left) of the 2D iris data (sepal length and width only). The solid ellipse represents how well the projection, onto different directions, separating the three species and the dashed circle is a reference guide. The histogram (right) is the 1D projection of the data onto the direction outlined as the solid line in the huber's plot.}
\label{fig:notation-huber}
\end{figure*}

## Display

Various displays are available to show projections in 1D, 2D, or higher dimensions. A 1D projection displays the data in a histogram as if a hand shadow being projected onto a wall. Two-d projections are usually shown as scatterplot since it is the most common display for bivariate relationship. Higher dimensional projections can be shown with multivariate display like scatterplot matrix or parallel coordinates plot. More display methods can be found in [@Wickham2011-uz]. Figure \ref{fig:display} shows a variety of displays for projections and these displays are useful to show the non-normal distributions in the projected data, clustering structure, and multivariate relationship between variables.

\begin{figure}
\centering
\includegraphics[width=0.3\textwidth]{figures/notation-display-1d.pdf}
\includegraphics[width=0.3\textwidth]{figures/notation-display-2d.pdf}
\includegraphics[width=0.3\textwidth]{figures/notation-display-5d.pdf}
\caption{Various displays of the projections: 1D display in histogram (left), 2D display in scatterplot (middle), and 5d display in parallel coordinate plot (right).}
\label{fig:display}
\end{figure}

## Finding targets 

To define whether a projection is interesting or not in the guided tour [EW: should we add a new section on tour types before this section?], an index function is used. An index function defines a surjection function from the projection $Y$ in $\mathbb{R}^{n \times d}$ to a scalar in $\mathbb{R}$. For the high-dimensional data, an interesting projection is the one that deviates from normality, as discussed in @Diaconis1984-mv and @Huber1985-zq [expand more here]. Numerous index functions have been developed since the initial index proposed by @Friedman1974-ck: <!-- entropy-based index? --> holes and central mass index [@Cook2008-nl] computes the deviation of the projected data from a normal distribution; @Posse1995-ih proposed a chi-square index for detecting the spiral structure in the data;  kurtosis and skewness indices are raised in @Loperfido2020-xj and @Loperfido2018-ga to detect outliers in financial time series; and most recently, scagnostic indices [@Laa2020-zj] provide eight measures for summarizing information between pairs of variables in SPLOM. 

\begin{figure}
\centering
\includegraphics[width=0.5\textwidth]{figures/notation-target-gen.pdf}
\caption{A comparison of guided and grand tour path in the PCA reduced space. The start point of tour paths is smaller and lighter than the end point. Guided tour optimises the index value iteratively and finishes the search quickly while grand tour wanders in the parameter space to look at possible interesting projections.}
\label{fig:notation-target-gen}
\end{figure}

## Geodesic interpolation

The smooth animation between the selected target frames is generated via a step-wise rotation in the high-dimensional space. Different methods to derive this rotation were described in @Buja2005-cx, and we can distinguish two types of interpolation [EW: ??]. When interpolating between frames we rotate the starting basis onto the exact basis representation selected as a basis, and @Buja2005-cx describes three different methods to achieve this (based on decompositions of orthogonal matrices, Givens decompositions, or Householder decompositions). Alternatively, when interpolating between planes the rotation is constructed towards any basis that represents the plane corresponding to the target frame, such that it follows the shortest path, i.e. the geodesic, avoiding any in-plane rotation. The geodesic interpolation is derived using singular value decomposition and is implemented in the `tourr` package. [EW: this paragraph perhaps needs some rework.]

Typically the geodesic interpolation is preferred, since within-plane rotation is distracting and does not provide new information. An exception might be the case of a projection pursuit guided tour with an index function that is sensitive to these rotations [@Laa2020-zj].

For any method we need to ensure that each projection along the interpolated path is still orthonormal. As explained above this means a tour path is a trace along a Stiefel manifold. In the case of one-dimensional projections, the set of bases traces a path along the surface of a hypersphere of radius one, as illustrated in Figure \ref{fig:notation-target-gen}, and for two-dimensional projections projections fall onto a torus shape. We can use a tour to visualize such a path, as shown in Figure \ref{fig:interpolation-2d}.

\begin{figure}
\centering
\includegraphics[width=0.9\textwidth]{figures/notation-2d-space.pdf}
\caption{Still views of a tour showing the space of possible two-dimensional projections as a torus shape in grey, and the trace of two short paths obtained via geodesic interpolation traced in color.}
\label{fig:interpolation-2d}
\end{figure}


## Slices, sections and projections

Interactive systems like GGobi provide the option of adding sectioning information via linked brushing. Sectioning means we select points that fall in a section of the full parameter space, for example for the purpose of highlighting them in a tour display. This often reveals complementary information, and combining sections and projections can for example provide insights into the dimensionality of a data structure [@Furnas1994-dz].

Sectioning a high-dimensional space in systematic manner (without interactive selection of data sections) is challenging, since there is a lot of freedom in choosing a section. One approach is to define sections based on projection planes, we refer to such sections as "slices" of the data. This is implemented in the slice tour display [@Laa2020-js], and uses the orthogonal distance of each data point from the current projection plane (typically placed such that it passes through the data mean) to highlight points that are nearby and fade out points further from the plane.

The slice tour can reveal concave or non-linear structures obscured in projections, as well as small structures hidden near the center of a distribution. Fig. \ref{fig:tour-slice} shows snapshots of the slice tour of points distributed on the surface of geometric shapes. Along with projections, an index function can be used to select target planes to show interesting slices of the data, defined as a section pursuit guided tour [@Laa2020-ni].

\begin{figure*}[ht]
\centerline{\includegraphics[width=0.2\textwidth]{figures/roman-surface-centered-010.png}
\hspace{5mm}
\includegraphics[width=0.2\textwidth]{figures/torus-4-centered-015.png}
\hspace{5mm}
\includegraphics[width=0.2\textwidth]{figures/cube-6-centered-001.png}}
\centerline{
\includegraphics[width=0.2\textwidth]{figures/roman-surface-centered-032.png}
\hspace{5mm}
\includegraphics[width=0.2\textwidth]{figures/torus-4-centered-033.png}
\hspace{5mm}
\includegraphics[width=0.2\textwidth]{figures/cube-6-centered-063.png}
}
\caption{Still views of the slice tour display showing geometric shapes: Roman surface (left), 4D torus (middle) and a 6d cube (right).}
\label{fig:tour-slice}
\end{figure*}


<!--
\begin{figure*}
   \centering
         \includegraphics[width=0.9\textwidth]{figures/slice_tour_left.jpg}
         \includegraphics[width=0.9\textwidth]{figures/slice_tour_right.jpg}
\caption{Ursula to write}
\label{fig:tour-slice}
\end{figure*}
-->


## Transformations on projections

Transformations of the data are common in the preprocessing [EW: "preprocessing" means b/f projecting?], such as scaling, sphering, or a logarithmic transformation for skewed distributions. However, we may also want to transform the data after projecting onto lower dimensions, to correct for unwanted effects of the projections. One example is the piling effect that is observed when projecting a high-dimensional distribution [@Diaconis1984-mv]: projected points are approximately Gaussian in most views, and increasingly concentrating near the center as dimensionality increases. The sage display [@Laa2020-uv] proposes a solution to this issue, via a radial transformation of the projected data. This non-linear transformation is defined in each point of the projection plane and is sensitive to the overall scale of the data as well as the original dimensionality $p$. In addition, tuning parameters can be used to obtain a more (or less) aggressive redistribution of the points. This is illustrated in Fig. \ref{fig:pollen} showing the application of the sage transformation to the classical pollen dataset [@Coleman1986-so]. The projection without any rescaling looks similar to what is found when rescaling with default options (left), and we can use either of the two tuning parameters to better resolve the distribution near the center and reveal the hidden structure (middle and right).

\begin{figure}
\centering
\includegraphics[width=0.3\textwidth]{figures/pollen_sage-088.png}
\includegraphics[width=0.3\textwidth]{figures/pollen_sage_R1-088.png}
\includegraphics[width=0.3\textwidth]{figures/pollen_sage_gam20-088.png}
\caption{Snapshots of the same projection from the sage tour on the pollen data, illustrating the different tuning options. Left: using the default settings for the transformation results in a similar view as found in projections without the sage transformation. Middle: using the tuning parameter $R$ for better resolution near the center reveals the word "EUREKA". Right: using the $\gamma$ tuning parameter can also reveal the structure.}
\label{fig:pollen}
\end{figure}

<!--
\begin{figure*}
   \centering
         \includegraphics[width=0.9\textwidth]{figures/laa_radial_transformation.pdf}
\caption{An illustration of the radial transformation. Each panel represents linear projections of $p$ dimensional equidistant circles. As $p$ increases the circles get pushed further to the edge.}
\label{fig:tour-transformation}
\end{figure*}
-->

# Ways of interacting {#interaction}

## Basic Interactions

Due to the dynamic nature of the tour, user interfaces can enhance the interpretability of the resulting visualisation. Often one will want to pause on a particularly interesting projection and return it for use in a downstream analysis or static plot.  Basic controls can be implemented so a tour display can be paused, refreshed and replayed over and over, as shown in figure \ref{fig:tour-controls}.

\begin{figure*}
   \centering
         \includegraphics[width=0.9\textwidth]{figures/liminal-stop-pause-refresh.png}
\caption{The clustering example from Figure \ref{fig:tsne-tour} using the liminal interface. Here a grand tour is displayed on the right hand side, with buttons allowing users to play, pause and refresh the tour animation.}
\label{fig:tour-controls}
\end{figure*}

## Manual tour 

<!-- needs to be shortened -->

Manual tours [@cook_manual_1997; @spyrison_spinifex_2020] offer a means to interactively control the contribution of a single variable on the projection plane. This is particularly useful for exploring a projection once a feature of interest has been identified. Manual tours can then be employed to explore local area of interest, or test the structure of the feature, with respect to a selected variable.

The details of performing a manual tour are described by @spyrison_spinifex_2020, but high-level unit-interaction between each frame is as follows:

__Input:__ Starting basis, manipulation variable, an in-plane rotation angle -- $\theta$, an out-of-plane rotation angle -- $\phi$  $\\$
__Output:__ Rotated basis $\\$
__Steps:__

1. Initialise a manipulation dimension onto the start basis. Given the manipulation variable a full contribution and orthonormalise this space. 
2. Post-multiply the space, by a rotation matrix, as solved for in application of Rodrigues’ formula [@rodrigues_lois_1840].
3. Truncate the manipulation dimension.

Figure \ref{fig:manual-tour} illustrates the manual tour, by radially manipulating the contribution of variable that is crucial, to explain the separation of a particular cluster. To start we first identify a basis by orthonormalising the linear discriminant. We select a variable that has a large contribution in the basis, and watch a continuous animation as this variable is rotated fully into the projection, completely out of the projection, and then returned to it's original contribution.

```{r manual-tour, out.width="100%", fig.show='asis', fig.cap='Manual tour of flea data. We initalise the start basis to the linear discriminant analysis and select the variable with the largest contribution separating the cluster shown as green circles. By controlling the contribution of the variable we see that it is quite sensitive to the sepration of this cluster.', eval=FALSE}
if(F) ## Creation and saving figure at:
  file.edit("./R/make_radial_tour_fig.r")
knitr::include_graphics("./figures/fig_radial_manual_tour.png")
```


\begin{figure*}[!ht]
   \centering
         \includegraphics[width=0.32\textwidth]{figures/penguins_manual_bl1_r.pdf}
         \includegraphics[width=0.32\textwidth]{figures/penguins_manual_bl3_r.pdf}
         \includegraphics[width=0.32\textwidth]{figures/penguins_manual_bl4_r.pdf}
\caption{Three projections from a manual tour where variable "bl" is being rotated out of the projection (orange line). When this variable is removed the two light green clusters merge, which informs us that "bl" is an important variable for distinguishing between these two groups.}
\label{fig:manual-tour}
\end{figure*}


## Linked Brushing

[EW: seems unrelated to the rest of the article; alternatively briefly explain "brushing" in s2.5?]

Brushing is an interaction that results in a region being drawn onto the view of visualisation. When there are multiple views present the act of brushing can be thought of as a database query; points that fall inside the brush can be used as a lookup to highlight or filter data on adjacent views (Figure \ref{fig:brushing}). The effect of brushing can be transient, that is, the changes triggered by a brush are removed after use, or they can be persistent i.e. the changes to data within the same view or adjacent views remain (Figure \ref{fig:prim7-tour}). Brushing can be used to aid statistical and geometrical interpretations of the data; brushing can be thought of as conditioning variables on certain regions of the data or used to section lower dimensional views.  

\begin{figure*}[!ht]
   \centering
         \includegraphics[width=0.45\textwidth]{figures/ggobi_book/brush-query.pdf}
         \includegraphics[width=0.45\textwidth]{figures/ggobi_book/brush-response.pdf}
\caption{When a user brushes data on the left panel, points that fall inside the region are highlighted in orange. The view on the right responds by highlighting the corresponding points.}
\label{fig:brushing}
\end{figure*}

## Spin-and-brush 

\begin{figure}[htp]
\centerline{{\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp1.pdf}}
 {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp2.pdf}}
 {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp5.pdf}}}
\smallskip
\centerline{{\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp7.pdf}}
 {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp8.pdf}}
 {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp9.pdf}}}
\centerline{{\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp10.pdf}}
  {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp11.pdf}}
  {\includegraphics[width=0.32\textwidth]{figures/ggobi_book/prim7-pp13.pdf}}}
\caption{Stepwise identification of clusters in the physics data. Each time a cluster is clearely separated we stop the tour and brush the points, followed by another spin of the data, until we can capture the full distribution.}
\label{fig:prim7-tour}
\end{figure}

Tasks such as the identification of outlying points or cluster analysis can be performed visually using the tour. Here, a "spin-and-brush" approach works well, since different views will reveal different features and a persistent brush helps us to connect this to the previously observed information. As an example we briefly summarize an analysis from @Cook2007-be that explores clustering of the physics data previously used to in the context of projection pursuit [@Friedman1974-ck; @Cook1995-ae]. This data is well described by its geometric structure: a two-dimensional triangle with two one-dimensional strands linearly extending in different directions from each vertex. These strands can sequentially be identified as clusters using the spin-and-brush approach, as shown in Figure \ref{fig:prim7-tour}. We stop the tour each time a cluster is clearly separate from the main distribution in the current projection and brush the points in a new color. In the end the full structure becomes apparent and can be visualized by replaying the tour with all clusters highlighted in different colors.



# Software {#software}

There has been a long history of software including tour methods, or something similar. These have laid the foundation for the current tools. The American Statistical Association Statistical Graphics Video Library web site [@ASASGVL] hosts videos documenting the history. It is fabulous watching the videos titled "Multidimensional Scaling" (Kruskal, 1962), "Real-Time Rotation" (Chang, 1970) and "Prim-9" (Tukey, 1973) show some preliminary methods leading to the development of the grand tour. The video titled "Use of the Grand Tour in Remote Sensing" (McDonald and Willis, 1987) is the first to show a tour, and the video titled "Dataviewer: A Program for Looking at Data in Several Dimensions" (Buja and Tukey, 1987) demonstrates the tour as part of a larger data analysis system. The videos titled "XGobi: Dynamic Graphics for Data Analysis" (Swayne et al, 1991) and "Grand Tour and Projection Pursuit" (Cook et al, 1993) show the tour tools in the XGobi software system. Some work on tours, for example @Wegman92 and @Tierney1991 is not documented in the video collection.

Recently, some other review papers have summarized different aspects of tour methods. @Moustafa2009 explains the relationship between a grand tour and the classical multivariate plot, the Andrews curve. @Moustafa2010 describes a computational shortcut for a tour. @Martinez2013 explains the image grand tour. 

The most accessible current software is the R package called tourr [@Wickham2011-uz]. This software evolved from the Dataviewer [@Buja1986-la], XGobi [@Swayne1998], GGobi [@Swayne2003], Orca [@Sutherland2000] and cranvas [@Xie2014] ancestry. The tourr package has a wide range of display types, for different projection dimensions, and a selection of target generating methods including grand, guided, little, local, section, sage and frozen. 

The R package spinifex [@spyrison_spinifex_2020] has 3 primary features: it produces manual tours (predefined path or interactive manipulation, identifies orthonormal global feature bases with the use of the Rdimtools [@you_rdimtools_2020] package, renders (manual or other) tours as animations exportable to static .gif (gganimate [@pedersen_gganimate_2020] package) or interactive .html widgets (plotly [@sievert_interactive_2020] package), lastly it offers interactive shiny application that offers an graphical user interface to quickly sample tour features.

The liminal R package uses the tour to explore the quality of non-linear dimensionality reduction algorithms [@Lee2020-kg]. The interface consists of two side by side views consisting of a scatter plot displaying a reduced form of the data, and an interactive tour. Controls such as play/stop/restart are implemented allowing a user to pause on interesting projections and return them to their R session for further analysis. Linked brushing is implemented on both views - if a users brush on the scatter plot view they can see if and how an algorithm like t-SNE has distorted distances in higher dimensional space, while if they brush on the tour view, the tour is paused and structure like cluster separation can be ascertained.      

- other work (Di)


# Applications {#applications}


## Model Visualization


\begin{figure*}
  \centering
    \includegraphics[width=0.32\textwidth]{figures/svm-radial-orange.png}
    \includegraphics[width=0.32\textwidth]{figures/svm-radial-green.png}
    \includegraphics[width=0.32\textwidth]{figures/svm-radial-purple.png}
  \caption{A grand tour run on the prediction boundaries of a radial support vector machine. (Left) The prediction region for points classified as orange produces a slice of a cube. (Middle) The predicition region for points classified as green produces another slice of the cube. (Right) The complete prediction region shows the points classified as purple are embedded within the green slice.}
  \label{fig:svm-tour}
\end{figure*}

@Wickham2015-do espouses the tour as a key component of their approach for visualizing models in data space. They propose using the tour to explore the high dimensional surfaces produced by model fits instead of simple summary statistics.  In Figure \ref{fig:svm-tour} samples have been taken along the prediction region (i.e. the value the model predicts over a grid of observations) of a 3-$d$ support vector machine used to classify three different classes (green, orange purple), by touring over the prediction region we can see how the classifier splits the decision boundaries.  

\begin{figure*}
   \centering
         \includegraphics[width=0.9\textwidth]{figures/scheidigger_nn_tour.png}
\caption{A grand tour run on the output of a deep neural network model for classification. The axes correspond to the probabilities that the model of an image belonging to a certain class.}
\label{fig:tour-nn}
\end{figure*}


The grand tour along with direct interaction techniques has been proposed as a technique for understanding the training of deep neural network models [@Li2020-kg]. They proposed a novel method for aligning the output of different layers by interpolating the output of different layers of a deep neural network via the the grand tour. They also apply the grand tour to explore the changes in single layers of the network for each training epoch. For example, by the running tour over the softmax layer outputs of a classification model, an analyst may understand where in the training confusion between classes occur and gain a view of model performance (Figure \ref{fig:tour-nn}). Similarly, the difference between training and test sets can be assessed through side by side linked tours of layer outputs. @li_toward_2020 extends this idea by first reducing the dimensionality of output layers with UMAP and then using the grand tour with manual controls over 15 dimensional embeddings to understand layer topology.


## Physics 

Physics models often consider $\mathcal{O}(10)$ free parameters and are compared to a much larger number of experimental observables ($\mathcal{O}(100)$). The comparison typically relies on numeric computation of the model predictions for all observables, potentially obscuring the nature of their dependence on the model parameters. Here, multivariate visualisation methods can provide new insights. This was demonstrated in @Cook2018-jm. For example, we explored grouping of experiments based on how they constrain the parameter space and we also identified an interesting multivariate outlier that pointed to potential issues with the data point. For illustration we show static views from the tour showing the orthogonal structure of the three groups (indicated by color), and a display which highlights the outlyingness of the data point marked with an asterisk symbol in Fig. \ref{fig:physics}.

\begin{figure*}
  \centering
           \includegraphics[width=0.45\textwidth]{figures/allcenter-301.png}
         \includegraphics[width=0.45\textwidth]{figures/jetCluster-027.png}
\caption{Selected frames from using a grand tour of the physics data. Left: the three different types of measurements (shown in different colors) are aligned along different directions in parameter space. Right: an outlying point is marked with an asterisk symbol in the ATLAS7new facet.}
\label{fig:physics}
\end{figure*}


## Bioinformatics

In single cell RNA sequencing, scientists are interested in identifying novel cell types and understanding the relationships between cells or their developmental trajectory. To achieve this they perform cluster analysis on a counts matrix or principal components and embed the results via t-SNE and label points according the cluster label. One of the main advantages of t-SNE is the avoidance of over-plotting so clusters can be clearly identified on a scatter plot, however, this can come at the cost of interpretability as global distances are distorted.  In @Laa2020-uv, we used radial transformations of the tour projections as an alternative to t-SNE that better preserves global structure while still retaining cluster topology.

\begin{figure*}
  \centering
           \includegraphics[width=0.24\textwidth]{figures/mouse_sage_2c_gam3-001.png}
         \includegraphics[width=0.24\textwidth]{figures/mouse_sage_2c_gam3-016.png}
         \includegraphics[width=0.24\textwidth]{figures/mouse_sage_2c_gam3-038.png}
         \includegraphics[width=0.24\textwidth]{figures/mouse_sage_2c_gam3-074.png}

\caption{Selected frames from using a grand tour of mouse retina single cell RNA-seq data with the radial transformation. The sage display was used to identify and verify cluster sepearation between the three highlighted clusters estimated from a clustering algorithm.}
\label{fig:mouse-cluster}
\end{figure*}


## Geometry of data

High-dimensional data is on many analysts' minds in recent years, and a way to build understanding of high-dimensional spaces is to examine common high-dimensional shapes using a tour. Many different shapes can be simulated using the R package geozoo. Figure \ref{fig:geozoo} shows points on the surface of two different types of high-dimensional torii. 

\begin{figure*}
  \centering
  \includegraphics[width=0.24\textwidth]{figures/torus-rings-4-with-torus-1.pdf}\includegraphics[width=0.24\textwidth]{figures/torus-rings-4-with-torus-2.pdf}
  \includegraphics[width=0.24\textwidth]{figures/torus-flat-6-1.pdf}\includegraphics[width=0.24\textwidth]{figures/torus-flat-6-2.pdf}
\caption{Several views of torii: (two at left) 4D ring torus, and (two at right) 6D flat torus. In each case, a subset has been highlighted which illustrate the low-dimensional donut shapes.}
\label{fig:geozoo}
\end{figure*}
          


# Discussion {#discussion}

* summarising the paper again
* highlighting the recent developments, spinifex, liminal, sage, slicing and sectioning, ferrn etc. 
* what's next - conditional model visualisation (Catherine Hurley's work), a interactive graphics system that easily integrates tours that provides the direct manipulations we want, pre-processing / transformations for high dimensional data, penalised projection pursuit, different kinds of data and spaces


# References {-}
